packages/salesforcedx-vscode-soql/test/jest/sfdx.test.ts

```diff
@@ -0,0 +1,54 @@
+/*
```

Comment:
@randi274 I added unit tests here. I spent way too long trying to make the `onOrgChangeDefaultHandler` test a true unit test and mock `withSFConnection` (its direct dependant) instead of `debouncedShowChannelAndErrorMessage`(two levels down), but Jest was just not liking it. So, now we have two unit tests that are largely similar. I'm fine keeping them both, killing one, or combing.

Reply:
Hmm... and [CI just ran](https://github.com/forcedotcom/salesforcedx-vscode/actions/runs/4952583307/jobs/8859073617) and failed on this:

```
TypeError: Cannot read properties of undefined (reading 'appendLine')
```

Looks like that is coming from an async timer calling that code outside of the test's mocks. That did not happen locally, but assuming it's a race condition because the timer had longer to run on CI. 

I'll dig more into this tomorrow.

Reply:
FYI I was able to repo locally using the following: 
```
ansible|~/github/PDT/salesforcedx-vscode (rb/soql-default-org-no-error ✔) node 'node_modules/.bin/jest' '/Users/gbockus/github/PDT/salesforcedx-vscode/packages/salesforcedx-vscode-soql/test/jest/sfdx.test.ts' -c '/Users/gbockus/github/PDT/salesforcedx-vscode/packages/salesforcedx-vscode-soql/jest.config.js' -t 'onOrgChangeDefaultHandler'
 PASS  packages/salesforcedx-vscode-soql/test/jest/sfdx.test.ts (5.203 s)
  withSFConnection
    ○ skipped should  show error message when showErrorMessage=true
    ○ skipped should not  show error message when showErrorMessage=false
  onOrgChangeDefaultHandler
    ✓ should  show connection error message when default org is  set (2 ms)
    ✓ should not  show connection error message when default org is not  set (1 ms)

Test Suites: 1 passed, 1 total
Tests:       2 skipped, 2 passed, 4 total
Snapshots:   0 total
Time:        5.318 s, estimated 6 s
Ran all test suites matching /\/Users\/gbockus\/github\/PDT\/salesforcedx-vscode\/packages\/salesforcedx-vscode-soql\/test\/jest\/sfdx.test.ts/i with tests matching "onOrgChangeDefaultHandler".
/Users/gbockus/github/PDT/salesforcedx-vscode/packages/salesforcedx-utils-vscode/out/src/commands/channelService.js:84
        this.channel.appendLine(text);
                     ^

TypeError: Cannot read properties of undefined (reading 'appendLine')
    at ChannelService.appendLine (/Users/gbockus/github/PDT/salesforcedx-vscode/packages/salesforcedx-utils-vscode/src/commands/channelService.ts:106:18)
    at showChannelAndErrorMessage (/Users/gbockus/github/PDT/salesforcedx-vscode/packages/salesforcedx-vscode-soql/src/sfdx.ts:18:28)
    at Timeout.later [as _onTimeout] (/Users/gbockus/github/PDT/salesforcedx-vscode/packages/salesforcedx-vscode-soql/node_modules/debounce/index.js:27:23)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
ansible|~/github/PDT/salesforcedx-vscode (rb/soql-default-org-no-error ✔) echo $?
1
ansible|~/github/PDT/salesforcedx-vscode (rb/soql-default-org-no-error ✔) 
```

Reply:
Awesome, thanks @gbockus-sf ! That worked for me now. Interesting that it doesn't happen with the `jest` CLI does does calling it from `node`.

Reply:
Fixed it up. The issue was that I was not setting `.mockImplementation(() => Promise.resolve())` so the real impl was getting called async.